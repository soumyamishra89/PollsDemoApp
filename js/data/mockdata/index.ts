export const mockQuestions = [
    {
        "question": "Favorite programming language?",
        "published_at": "2020-04-15T14:49:59.810110+00:00",
        "url": "/questions/5",
        "choices": [
            {
                "choice": "Objective-C",
                "votes": 0,
                "url": "/questions/5/choices/19"
            },
            {
                "choice": "Python",
                "votes": 0,
                "url": "/questions/5/choices/18"
            },
            {
                "choice": "Ruby",
                "votes": 0,
                "url": "/questions/5/choices/20"
            },
            {
                "choice": "Swift",
                "votes": 0,
                "url": "/questions/5/choices/17"
            }
        ],
        "totalVotes": 0
    },
    {
        "question": "Capital of Germany?",
        "published_at": "2020-04-15T11:00:59.542382+00:00",
        "url": "/questions/4",
        "choices": [
            {
                "choice": "Berlin",
                "votes": 0,
                "url": "/questions/4/choices/13"
            },
            {
                "choice": "Frankfurt",
                "votes": 0,
                "url": "/questions/4/choices/16"
            },
            {
                "choice": "Hamburg",
                "votes": 0,
                "url": "/questions/4/choices/14"
            },
            {
                "choice": "Munich",
                "votes": 0,
                "url": "/questions/4/choices/15"
            }
        ],
        "totalVotes": 0
    },
    {
        "question": "test q?",
        "published_at": "2020-04-15T09:59:32.552020+00:00",
        "url": "/questions/3",
        "choices": [
            {
                "choice": "Objective-C",
                "votes": 0,
                "url": "/questions/3/choices/11"
            },
            {
                "choice": "Python",
                "votes": 0,
                "url": "/questions/3/choices/10"
            },
            {
                "choice": "Ruby",
                "votes": 0,
                "url": "/questions/3/choices/12"
            },
            {
                "choice": "Swift",
                "votes": 0,
                "url": "/questions/3/choices/9"
            }
        ],
        "totalVotes": 0
    },
    {
        "question": "",
        "published_at": "2020-04-10T16:05:25.127319+00:00",
        "url": "/questions/2",
        "choices": [
            {
                "choice": "Python",
                "votes": 14,
                "url": "/questions/2/choices/6"
            },
            {
                "choice": "Ruby",
                "votes": 6,
                "url": "/questions/2/choices/8"
            },
            {
                "choice": "Objective-C",
                "votes": 2,
                "url": "/questions/2/choices/7"
            },
            {
                "choice": "Swift",
                "votes": 1,
                "url": "/questions/2/choices/5"
            }
        ],
        "totalVotes": 23
    },
    {
        "question": "Favourite programming language?",
        "published_at": "2020-04-09T03:28:20.545996+00:00",
        "url": "/questions/1",
        "choices": [
            {
                "choice": "Python",
                "votes": 34,
                "url": "/questions/1/choices/2"
            },
            {
                "choice": "Objective-C",
                "votes": 8,
                "url": "/questions/1/choices/3"
            },
            {
                "choice": "Ruby",
                "votes": 4,
                "url": "/questions/1/choices/4"
            },
            {
                "choice": "Swift",
                "votes": 3,
                "url": "/questions/1/choices/1"
            }
        ],
        "totalVotes": 49
    }
]
