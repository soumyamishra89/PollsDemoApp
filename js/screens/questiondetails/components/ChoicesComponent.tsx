import React from 'react';
import { RadioButton as RadioButtonPaper } from 'react-native-paper';
import { PollChoice } from '../../../data/types/PollsData';
import RadioButton from '../../common/compoonent/RadioButton';
import utils from '../../../services/utils';

interface Props {
    totalVotes: number
    choices: PollChoice[],
    selectedChoice?: PollChoice,
    onChoiceChange: (choice: PollChoice) => void
}
export default function ChoicesComponent(props: Props) {
    
    const onChoiceChange = (choice: string) => {
        const selectedChoice = props.choices.find(c => c.choice === choice);
        selectedChoice && props.onChoiceChange(selectedChoice);
    }
    return (       
        <RadioButtonPaper.Group 
        onValueChange={onChoiceChange}
        value={props.selectedChoice?.choice || ''}>
            {props.choices.map(choice => <RadioButton 
            key={choice.url}
            isActive={props.selectedChoice?.choice === choice.choice} 
            value={choice.choice} 
            subtext={utils.getPercentageAsString(choice.votes, props.totalVotes)}
            activePercentWidth={utils.getPercentageAsString(choice.votes, props.totalVotes)}/>)}
        </RadioButtonPaper.Group>
    )
}