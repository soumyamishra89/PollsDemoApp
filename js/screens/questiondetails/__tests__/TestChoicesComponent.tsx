import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ChoicesComponent from '../components/ChoicesComponent';
import { mockQuestions } from '../../../data/mockdata';

describe("ChoicesComponent", () => {
    describe("Snapshot test", () => {
        it("With no choice selected", () => {
            const pollQuestion = mockQuestions[0];
            const choicesTree = renderer.create(<ChoicesComponent totalVotes={pollQuestion.totalVotes}
                choices={pollQuestion.choices}
                onChoiceChange={jest.fn()} />).toJSON();
            expect(choicesTree).toMatchSnapshot();

        })

        it("With choice selected", () => {
            const pollQuestion = mockQuestions[0];
            const choicesTree = renderer.create(<ChoicesComponent totalVotes={pollQuestion.totalVotes}
                choices={pollQuestion.choices}
                selectedChoice={pollQuestion.choices[0]}
                onChoiceChange={jest.fn()} />).toJSON();
            expect(choicesTree).toMatchSnapshot();

        })
    })
})