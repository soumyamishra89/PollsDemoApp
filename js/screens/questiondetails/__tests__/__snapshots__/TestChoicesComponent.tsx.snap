// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoicesComponent Snapshot test With choice selected 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          Array [
            Object {
              "backgroundColor": "#B2CBFFAA",
            },
            Object {
              "width": "0.00%",
            },
          ],
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#3C7AFF",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#3C7AFF",
                  "borderRadius": 5,
                  "height": 10,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              Object {
                "color": "#3C7AFF",
              },
            ],
          ]
        }
      >
        Objective-C
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            Object {
              "color": "#3C7AFF",
            },
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Python
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Ruby
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Swift
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
]
`;

exports[`ChoicesComponent Snapshot test With no choice selected 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Objective-C
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Python
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Ruby
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#F1F5FF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 24,
            "height": "100%",
            "position": "absolute",
          },
          false,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#BFD3FE",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              undefined,
              false,
            ],
          ]
        }
      >
        Swift
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "marginRight": 8,
            },
            false,
          ],
        ]
      }
    >
      0.00%
    </Text>
  </View>,
]
`;
